packages/compass-e2e-tests/tests/assistant.test.ts

```diff
@@ -0,0 +1,581 @@
+          expect(finalMessages.length).to.equal(0);
+
+          expect(await chatMessages.getText()).to.include(
+            'Please confirm your request'
+          );
+          expect(await chatMessages.getText()).to.include('Request cancelled');
+
+          // Verify no assistant request was made
+          expect(mockAssistantServer.getRequests()).to.be.empty;
+        });
+      });
+
+      describe('error message entry point', function () {
+        before(async function () {
+          await setAIOptIn(browser, true);
+
+          mockAssistantServer.setResponse({
+            status: 200,
+            body: 'You should review the connection string.',
+          });
+        });
+
+        it('opens assistant with error message view prompt when clicking "Debug for me"', async function () {
+          await browser.connectWithConnectionString(
+            'mongodb-invalid://localhost:27017',
+            { connectionStatus: 'failure' }
+          );
+          await useErrorViewEntryPoint(browser);
+
+          const messages = await getDisplayedMessages(browser);
+          expect(messages).deep.equal([
+            {
+              text: 'Diagnose why my Compass connection is failing and help me debug it.',
+              role: 'user',
+            },
+            {
+              text: 'You should review the connection string.',
+              role: 'assistant',
+            },
+          ]);
+
+          expect(mockAssistantServer.getRequests()).to.have.lengthOf(1);
+        });
+      });
+    });
+  });
+});
+
+async function setAIFeatures(browser: CompassBrowser, newValue: boolean) {
+  await browser.openSettingsModal('ai');
+
+  // Wait for AI settings content to be visible
+  const aiSettingsContent = browser.$(
+    Selectors.ArtificialIntelligenceSettingsContent
+  );
+  await aiSettingsContent.waitForDisplayed();
+
+  const currentValue =
+    (await browser
+      .$(Selectors.SettingsInputElement('enableGenAIFeatures'))
+      .getAttribute('aria-checked')) === 'true';
+
+  if (currentValue !== newValue) {
+    await browser.clickParent(
+      Selectors.SettingsInputElement('enableGenAIFeatures')
+    );
+    await browser.clickVisible(Selectors.SaveSettingsButton);
+  }
+
+  const closeButton = browser.$(Selectors.CloseSettingsModalButton);
+  await closeButton.waitForClickable();
+  await closeButton.click();
+  await closeButton.waitForDisplayed({
+    reverse: true,
+  });
+}
+
+async function setAIOptIn(browser: CompassBrowser, enabled: boolean) {
+  // Reset the opt-in preference by using the execute command
+  await browser.setFeature('optInGenAIFeatures', enabled);
+
+  // Wait for the IPC to be processed
+  await browser.pause(500);
+}
+
+async function openAssistantDrawer(browser: CompassBrowser) {
+  const drawerButton = browser.$(Selectors.AssistantDrawerButton);
+  await drawerButton.waitForDisplayed();
+  await drawerButton.waitForClickable();
+  await drawerButton.click();
```

Comment:
Just FYI in theory `await browser.clickVisible(Selectors.AssistantDrawerButton);` does all this and more.

Would have been better if we added it as a custom element command rather than a browser command, but it is there.